apis/r/tests/testthat/test-Timestamps.R

```diff
@@ -0,0 +1,108 @@
+test_that("SOMADataFrame", {
+    uri <- tempfile()
+
+    sch <- arrow::schema(arrow::field("soma_joinid", arrow::int64()),
+                         arrow::field("int", arrow::int32()),
+                         arrow::field("str", arrow::dictionary(index_type = arrow::int8(),
+                                                               value_type = arrow::utf8())))
+
+    ## create at t = 1
+    ts1 <- rep(as.POSIXct(1, tz="UTC"), 2)
```

Comment:
Marking once for the entire file:
- `tiledb_timestamp` should be a scalar indicating "read up to this point"
- `$write()` and `$read*()` should not accept a timestamp, instead, we should `$reopen()` at the desired timestamp for writing at/reading from new timestamps

Reply:
Scalar vs vector addressed in #2929

Reply:
`$write()` addressed in #2930